golos.config/include/golos.config/parameters.hpp

```diff
@@ -15,6 +15,11 @@ struct base_threshold: parameter {
         // TODO: must check value realted to max_witnesses
         eosio_assert(value > 0, (std::string(*N) + " can't be 0").c_str());
     }
+
+    template<typename T>
+    bool compare(const T& other) const {
```

Comment:
This is a bad practice, forcing the programmer to write the code of the comparison function in each parameter. It is correct to write once the code of the comparison function, which performs the default comparison, and only in the some places replace this function.